compiler/src/iree/compiler/Codegen/Common/TransformExtensions/CommonExtensions.cpp

```diff
@@ -696,5 +588,242 @@ transform_dialect::TileToForeachThreadAndWorkgroupCountRegion::apply(
+    ValueRange dynamicSizes, unsigned alignment) {
+  // TODO: use gpu::GPUDialect::getWorkgroupAddressSpace() but this requires
+  // moving out of CommonExtensions.
+  MemRefType allocType = MemRefType::get(memRefType.getShape(),
+                                         memRefType.getElementType(), {}, 3);
+  return builder
+      .create<memref::AllocOp>(loc, allocType, dynamicSizes,
+                               builder.getI64IntegerAttr(alignment))
+      .getResult();
+}
+
+static LogicalResult gpuComprehensiveBufferizeDeallocationFn(OpBuilder &builder,
+                                                             Location loc,
+                                                             Value allocation) {
+  builder.create<memref::DeallocOp>(loc, allocation);
+  return success();
+}
+
+static LogicalResult gpuComprehensiveBufferizeCopyFn(OpBuilder &builder,
+                                                     Location loc, Value from,
+                                                     Value to) {
+  // TODO: ideally we should use linalg.copy which was recently reintroduced
+  // as an OpDSL named op. However, IREE-specific patterns to cleanup spurious
+  // post-bufferization copies do not trigger properly.
+  // So we keep using `createLinalgCopyOp` which builds a GenericOp.
+  // builder.create<linalg::CopyOp>(loc, from, to);
+  mlir::iree_compiler::createLinalgCopyOp(builder, loc, from, to);
+  return success();
+}
+
+/// Temporarily copied from IREEComprehensiveBufferizePass.cpp to avoid buying
+/// into nested pass pipeline mess.
+static LogicalResult emptyTensorElimination(
+    Operation *op, OneShotBufferizationOptions options) {
+  // Analyze IR.
+  options.testAnalysisOnly = false;
+  options.printConflicts = false;
+  OneShotAnalysisState state(op, options);
+  if (failed(analyzeOp(op, state))) return failure();
+
+  // Rewrite init_tensors that are anchored on specific ops.
+  IRRewriter rewriter(op->getContext());
+  if (failed(bufferization::insertSliceAnchoredAllocTensorEliminationStep(
+          rewriter, op, state)))
+    return failure();
+  if (failed(
+          storeTensorOpAnchoredInitTensorEliminationStep(rewriter, op, state)))
+    return failure();
+
+  return success();
+}
+
+/// Temporarily copied from IREEComprehensiveBufferizePass.cpp to avoid buying
+/// into nested pass pipeline mess.
+// The following is copied from bufferization::runOneShotBufferize with
+// modifications.
+static LogicalResult runIREEOneShotBufferize(
+    Operation *op, const OneShotBufferizationOptions &options) {
+  OneShotAnalysisState state(op, options);
+  if (failed(analyzeOp(op, state))) return failure();
+  if (options.testAnalysisOnly) return success();
+  return bufferization::runOneShotBufferize(op, options);
+}
+
+/// Temporarily copied from IREEComprehensiveBufferizePass.cpp to avoid buying
+/// into nested pass pipeline mess.
+static LogicalResult runOnModule(
```

Comment:
Should be prefixed with with `bufferize` etc. as there are multiple transform ops in this file.